src/checkers/inference/solver/frontend/TwoQualifiersLattice.java

```diff
@@ -17,10 +19,23 @@
     public TwoQualifiersLattice(Map<AnnotationMirror, Collection<AnnotationMirror>> subType,
             Map<AnnotationMirror, Collection<AnnotationMirror>> superType,
             Map<AnnotationMirror, Collection<AnnotationMirror>> incomparableType,
-            Map<AnnotationMirror, Integer> typeToInt, Map<Integer, AnnotationMirror> intToType,
             Set<? extends AnnotationMirror> allTypes, AnnotationMirror top, AnnotationMirror bottom,
             int numTypes) {
-        super(subType, superType, incomparableType, typeToInt, intToType, allTypes, top, bottom,
-                numTypes);
+        super(subType, superType, incomparableType, allTypes, top, bottom, numTypes, null, null);
```

Comment:
should refactor super constructor to not require the 2 null arguments. Maybe an abstract Lattice and 2 concrete lattice classes.

Reply:
Good suggestion. Let me think about how to refactor this.